lib/metasploit/framework/ldap/client.rb

```diff
@@ -0,0 +1,141 @@
+              host: opts[:domain_controller_rhost],
+              hostname: opts[:ldap_rhostname],
+              realm: opts[:domain],
+              username: opts[:username],
+              password: opts[:password],
+              framework: opts[:framework],
+              framework_module: opts[:framework_module],
+              cache_file: opts[:ldap_krb5_cname].blank? ? nil : opts[:ldap_krb5_cname],
+              ticket_storage: opts[:kerberos_ticket_storage],
+              offered_etypes: offered_etypes
+            )
+
+            connect_opts[:auth] = {
+              method: :sasl,
+              mechanism: 'GSS-SPNEGO',
+              initial_credential: proc do
+                kerberos_result = kerberos_authenticator.authenticate
+                kerberos_result[:security_blob]
+              end,
+              challenge_response: true
+            }
+          when Msf::Exploit::Remote::AuthOption::NTLM
+            ntlm_client = RubySMB::NTLM::Client.new(
+              opts[:username],
+              opts[:password],
+              workstation: 'WORKSTATION',
+              domain: opts[:domain].blank? ? '.' : opts[:domain],
+              flags:
+                RubySMB::NTLM::NEGOTIATE_FLAGS[:UNICODE] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:REQUEST_TARGET] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:NTLM] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:ALWAYS_SIGN] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:EXTENDED_SECURITY] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:KEY_EXCHANGE] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:TARGET_INFO] |
+                  RubySMB::NTLM::NEGOTIATE_FLAGS[:VERSION_INFO]
+            )
+
+            negotiate = proc do |challenge|
+              ntlmssp_offset = challenge.index('NTLMSSP')
+              type2_blob = challenge.slice(ntlmssp_offset..-1)
+              challenge = [type2_blob].pack('m')
+              type3_message = ntlm_client.init_context(challenge)
+              type3_message.serialize
+            end
+
+            connect_opts[:auth] = {
+              method: :sasl,
+              mechanism: 'GSS-SPNEGO',
+              initial_credential: ntlm_client.init_context.serialize,
+              challenge_response: negotiate
+            }
+          when Msf::Exploit::Remote::AuthOption::PLAINTEXT
+            # username = opts[:username].dup
+            # username << "@#{opts[:domain]}" unless opts[:domain].blank?
+            connect_opts[:auth] = {
+              method: :simple,
+              username: opts[:username],
+              password: opts[:password]
+            }
+          when Msf::Exploit::Remote::AuthOption::AUTO
+            unless opts[:username].blank? # plaintext if specified
+              # username = opts[:username].dup
+              # username << "@#{opts[:domain]}" unless opts[:domain].blank?
```

Comment:
Is this code meant to be commented out? 👀

Reply:
I believe that's just left over from when I moved the code, should be able to remove it